File: Domain/BeSpoked.BLL/Customers/Entities/Customer.cs
using BeSpoked.Common.EntityService.Entities;
using BeSpoked.Sales.Entities;

namespace BeSpoked.Customers.Entities;

public record Customer : Entity
{
    public required string FirstName { get; set; }
    public required string LastName { get; set; }
    //public Address
    public required string Phone { get; set; }
    public DateTime StartDate { get; set; }
    public string FullName => $"{FirstName} {LastName}";
    public virtual ICollection<Sale> Sales { get; set; } = new HashSet<Sale>();
};

File: WebApp/ClientApp/src/Entities/Customer.ts
export interface Customer {
    id: string;
    firstName: string;
    lastName: string;
    phone: string;
    startDate: Date;
}

export type ServerCustomer = Omit<Customer, "startDate"> & { startDate: string}

export function deserializeCustomer(serverResponse:ServerCustomer):Customer{
    return {
        ...serverResponse,
        startDate: new Date(serverResponse.startDate),
    }
}